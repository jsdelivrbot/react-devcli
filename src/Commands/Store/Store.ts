import {Arguments} from 'yargs';
import ACommand from '../ACommand';
import {CommandsConfig} from 'react-devcli';
import {IMakeable, IRemovable} from '../Commands';

class Store extends ACommand implements IRemovable, IMakeable {
    public constructor() {
        const name = 'store';
        const description = 'store description';
        const options = {};

        super(name, description, options);
    }

    public run(argv: Arguments, config: CommandsConfig): void {
        console.log(config);
        console.log(argv);
    }

    public help(argv: Arguments): void {
        console.log(argv);
    }

    public remove(argv: Arguments): void {
        console.log(argv);
    }

    public make(argv: Arguments): void {
        console.log(argv);
    }
}

export default Store;
