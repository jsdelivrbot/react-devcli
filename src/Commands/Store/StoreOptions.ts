export const name = 'store';

export const description = 'store description';

export const flags = {};
