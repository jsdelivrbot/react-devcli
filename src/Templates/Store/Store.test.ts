import {js, ts} from './Store';

describe('Store template test suite', (): void => {
    test('should match store template snapshot for JS language', (): void => {
        expect(js()).toMatchSnapshot();
    });

    test('should match store template snapshot for TS language', (): void => {
        expect(ts()).toMatchSnapshot();
    });
});
