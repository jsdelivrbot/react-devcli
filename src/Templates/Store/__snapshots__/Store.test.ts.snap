// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store template test suite should match store template snapshot for JS language 1`] = `""`;

exports[`Store template test suite should match store template snapshot for TS language 1`] = `""`;
